src/server/utils/bucket_diff.js

```diff
@@ -69,38 +69,47 @@ class BucketDiff {
             //     ]
             // }
             keys_diff_map: {},
+            keys_del_map: {},
             first_bucket_cont_token: '',
             second_bucket_cont_token: '',
         };
 
         let first_bucket_contents_left;
         let first_bucket_cont_token;
+        let first_bucket_delete_markers
         //list the objects in the first bucket
         ({
             bucket_contents_left: first_bucket_contents_left,
-            bucket_cont_token: first_bucket_cont_token
+            bucket_cont_token: first_bucket_cont_token,
+            delete_markers: first_bucket_delete_markers
         } = await this.get_objects(this.first_bucket, prefix, max_keys, current_first_bucket_cont_token));
 
         if (first_bucket_cont_token) diff.first_bucket_cont_token = first_bucket_cont_token;
         if (Object.keys(first_bucket_contents_left).length === 0) return diff;
 
         let second_bucket_contents_left;
         let second_bucket_cont_token;
+        let second_bucket_delete_markers;
         let new_second_bucket_cont_token = current_second_bucket_cont_token;
         let keep_listing_second_bucket = true;
 
         while (keep_listing_second_bucket) {
             ({
                 bucket_contents_left: second_bucket_contents_left,
-                bucket_cont_token: new_second_bucket_cont_token
+                bucket_cont_token: new_second_bucket_cont_token,
+                delete_markers: second_bucket_delete_markers
             } = await this.get_objects(this.second_bucket, prefix, max_keys, new_second_bucket_cont_token));
 
             const keys_diff_response = await this.get_keys_diff(
                 first_bucket_contents_left, second_bucket_contents_left, new_second_bucket_cont_token);
 
+            const delete_keys = await this.get_keys_del(
```

Comment:
Why do we need to call it directly from here and not from the `get_keys_diff` function?

Reply:
Sure, will update it.